Sources/TensorFlow/Core/DifferentialOperators.swift

```diff
@@ -29,7 +29,8 @@ public extension Differentiable {
         in f: @differentiable (Self) -> Tensor<R>
     ) -> (value: Tensor<R>, gradient: TangentVector) {
         let (y, pb) = self.valueWithPullback(in: f)
-        precondition(y.rank == 0)
+        precondition(y.rank == 0,
+            "The forward pass computed a non-scalar tensor (shape: \(y.shape)).")
```

Comment:
I would rather say something like "The function being differentiated produced a tensor with shape \(y.shape). Only scalar-valued functions can be differentiated." It's a bit more verbose but it should be easier to understand for first-time users.

Reply:
Thank you, that's much better! I've tweaked it a bit, however, because you can differentiate non-scalar-valued functions, it's just about whether you can compute the gradient for them. PTAL?